src/main/java/com/epam/third/task/observer/SphereObserver.java

```diff
@@ -0,0 +1,58 @@
+package com.epam.third.task.observer;
+
+import com.epam.third.task.entities.ObservableSphere;
+import com.epam.third.task.entities.SphereParameters;
+import com.epam.third.task.logic.SphereCalculator;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class SphereObserver implements Observer {
+    private final Map<Integer, SphereParameters> observableSpheres = new HashMap<>();
+    private SphereCalculator calculator;
+
+    private final static SphereObserver INSTANCE = new SphereObserver();
+
+    private SphereObserver() {
+    }
+
+    public static SphereObserver getInstance() {
+        return INSTANCE;
+    }
+
+    public void setCalculator(SphereCalculator calculator) {
```

Comment:
no setter here, just create it in the constructor, because until you set it your singleton is invalid